config/routes.rb

```diff
@@ -5,15 +5,26 @@
   # You can have the root of your site routed with "root"
   root 'task#index'
 
-  get "/task_:id/details" => "task#show"
+  post "/task_:id/mark_complete"   => "task#mark_complete"
+  post "/task_:id/mark_incomplete" =>      "task#mark_incomplete"
```

Comment:
Nice work creating new actions for a task member. In this case you _could_ have used the HTTP variables `POST` and `DELETE` to change the state. If you think of the completion time (or even the state of being complete) as a sub-resource on the task, then `POST` creates a completion date and `DELETE` removes it. But that's pretty advanced and this is a perfectly understandable (and perhaps more understandable) interface.